assume/markets/clearing_algorithms/contracts.py

```diff
@@ -153,6 +154,14 @@ def clear(
             demand_orders = list(filter(lambda x: x["volume"] < 0, product_orders))
             supply_orders = list(filter(lambda x: x["volume"] > 0, product_orders))
 
+            # Sort supply orders by price with randomness for tie-breaking
+            supply_orders.sort(key=lambda x: (x["price"], random.random()))
+
+            # Sort demand orders by price in descending order with randomness for tie-breaking
+            demand_orders.sort(
+                key=lambda x: (x["price"], random.random()), reverse=True
+            )
+
```

Comment:
I would rather place the addition of the second sorting criteria in the sort right below - otherwise we are sorting twice.
Same below

Reply:
@maurerle do you mean like this? demand_orders.sort(key=lambda x: (x["price"], random.random()), reverse=True) as a tuple of sorting keys or what?

Reply:
see #389